Lib/platform.py

```diff
@@ -818,6 +842,14 @@ def get_OpenVMS():
             csid, cpu_number = vms_lib.getsyi('SYI$_CPU', 0)
             return 'Alpha' if cpu_number >= 128 else 'VAX'
 
+    # On the iOS simulator, os.uname returns the architecture as uname.machine.
+    # On device it returns the model name for some reason; but there's only one
+    # CPU architecture for iOS devices, so we know the right answer.
+    def get_ios():
+        if sys.implementation._multiarch.endswith("simulator"):
+            return os.uname().machine
+        return 'arm64'
```

Comment:
```suggestion
        if hasattr(sys.implementation, '_multiarch') and sys.implementation._multiarch.endswith("simulator"):
            return os.uname().machine
        return 'arm64'
```

Reply:
Confirming: iOS builds will always have `_multiarch` set (because of [this section of configure.ac](https://github.com/python/cpython/blob/main/configure.ac#L1087)); and this block of logic will only be invoked on iOS.

Reply:
That is true for CPython, but it may change for different implementations that keep our stdlib. It's not a big deal, but generally I avoid adding a hard dependency on implementation details. This is certainly not a blocker, just a possible improvement.

Reply:
Avoiding hard dependencies on implementation details definitely makes sense; However, I'm not sure if using a fallback is entirely safe. Without any additional changes, using a fallback value for `_multiarch` will result in:

* All iOS code identifying as ARM64 on a physical device, regardless whether it is or not
* Attempting to load binary modules with the tag `ios-12.0-`, regardless of architecture or simulator status

The second point would at least complicate, and possibly prevent any binary module from loading; the first would lead to misleading or incorrect behavior. 

On that basis, I'd argue it would be safer to fail early, providing a clear indicator that something isn't configured correctly (or, at least, not configured the way that the CPython implementation expects). However, it's also a relatively straightforward change, so I'm happy to be overruled on this.